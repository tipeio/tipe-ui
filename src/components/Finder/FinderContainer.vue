<template>
  <type-finder-empty v-if="empty" />
  <type-finder-viewer
    v-else
    :files="files"
    :layout="layout"
    :icon="icon"
    @select="file => $emit('select', file)"
  />
</template>

<script>
import vueTypes from 'vue-types'
import interfaces from '@tipe/tipe-interfaces'
import TypeFinderViewer from './FinderViewer.vue'
import TypeFinderEmpty from './FinderEmpty.vue'

export default {
  name: 'TipeFinderContainer',
  components: {
    TypeFinderViewer,
    TypeFinderEmpty
  },
  props: {
    files: vueTypes.arrayOf(vueTypes.shape(interfaces.file)).def([]),
    layout: vueTypes.oneOf(['list', 'grid']).def('list'),
    icon: vueTypes.oneOf(['rectangle', 'square', 'details']).def('rectangle')
  },
  computed: {
    empty() {
      return this.files.length < 1
    }
  }
}
</script>
