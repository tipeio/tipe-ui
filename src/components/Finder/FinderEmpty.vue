<template>
  <div data-tipe-ui="TipeFinder">Empty</div>
</template>

<style lang="postcss" scoped>
[data-tipe-ui='TipeFinder'] {
  height: 100%;
  width: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
  font-size: 0.8125rem;
  color: var(--text-gray-normal);
  background-color: var(--gray-lightest);
}
</style>
