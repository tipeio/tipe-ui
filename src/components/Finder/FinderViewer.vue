<template>
  <div
    :data-tipe-ui="$options.name"
    class="viewer"
  >
    <div class="scrollable">
      <div :class="layoutClass">
        <type-file-icon
          v-for="(file, i) in files"
          :key="i"
          :file="file"
          class="item"
        />
      </div>
    </div>
  </div>
</template>

<script>
import vueTypes from 'vue-types'
import FileType from './FileType'

import TypeFileIcon from './FileIcon.vue'

export default {
  name: 'TipeFinder',
  components: {
    TypeFileIcon
  },
  props: {
    files: vueTypes.arrayOf(FileType).def([]),
    layout: vueTypes.oneOf(['list', 'grid']).def('list')
  },
  computed: {
    layoutClass() {
      return {
        layout: true,
        [this.layout]: true
      }
    }
  }
}
</script>

<style lang="postcss" scoped>
.viewer {
  overflow: hidden;
}

/* hide scrollbar */
.scrollable {
  width: 100%;
  height: 100%;
  overflow-y: scroll;
  padding-right: 17px;
  box-sizing: content-box;
}

.layout.list {
  display: grid;
  grid-template-columns: 1fr;
  grid-auto-rows: max-content;
  grid-gap: 0.8rem;
}

.layout.grid {
  display: grid;
  grid-template-columns: repeat(auto-fill, minmax(11rem, 1fr));
  grid-auto-rows: max-content;
  grid-gap: 0.8rem;
}
</style>
