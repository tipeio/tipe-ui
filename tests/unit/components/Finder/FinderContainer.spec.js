import { shallowMount } from '@vue/test-utils'
import FinderContainer from '@/components/Finder/FinderContainer'
import FinderEmpty from '@/components/Finder/FinderEmpty.vue'
import FinderViewer from '@/components/Finder/FinderViewer.vue'
import { mocks } from '@tipe/tipe-test-utils'

describe('FinderContainer', () => {
  describe('<template>', () => {
    it('matches previous snapshot', () => {
      const wrapper = shallowMount(FinderContainer)
      expect(wrapper).toMatchSnapshot()
    })
  })

  describe(':props', () => {
    it(':files(empty) - should render FinderEmpty componet', () => {
      const wrapper = shallowMount(FinderContainer)

      expect(wrapper.find(FinderEmpty).exists()).toBe(true)
    })

    it(':files(not empty) - should render FinderViewer component', () => {
      const propsData = { files: [mocks.file()] }
      const wrapper = shallowMount(FinderContainer, { propsData })

      expect(wrapper.find(FinderViewer).exists()).toBe(true)
    })

    it(':layout(grid) - should pass layout prop to FinderViewer component', () => {
      const propsData = { files: [mocks.file()], layout: 'grid' }
      const wrapper = shallowMount(FinderContainer, { propsData })

      expect(wrapper.find(FinderViewer).props().layout).toBe('grid')
    })

    it(':layout - should default to list', () => {
      const propsData = { files: [mocks.file()] }
      const wrapper = shallowMount(FinderContainer, { propsData })

      expect(wrapper.find(FinderViewer).props().layout).toBe('list')
    })

    it(':picker - should pass to FinderViewer component', () => {
      const propsData = { picker: true, files: [mocks.file()] }
      const wrapper = shallowMount(FinderContainer, { propsData })

      expect(wrapper.find(FinderViewer).props().picker).toEqual(
        propsData.picker
      )
    })

    it(':multi - should pass to FinderViewer component', () => {
      const propsData = { multi: true, files: [mocks.file()] }
      const wrapper = shallowMount(FinderContainer, { propsData })

      expect(wrapper.find(FinderViewer).props().multi).toEqual(propsData.multi)
    })
  })
})
