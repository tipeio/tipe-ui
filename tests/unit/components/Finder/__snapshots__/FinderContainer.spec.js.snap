// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FinderContainer <template> matches previous snapshot 1`] = `<typefinderempty-stub></typefinderempty-stub>`;
